File: backend/Model/CustomsDeclaration.cs
namespace CosmoCargo.Model
{
    public class CustomsDeclaration
    {
        public int Id { get; set; }
        public Guid ShipmentId { get; set; }
        
        public bool ContainsLifeforms { get; set; }
        public string? LifeformType { get; set; }
        public bool IsPlasmaActive { get; set; }
        public int? PlasmaStabilityLevel { get; set; }
        public bool QuarantineRequired { get; set; }
        public bool OriginPlanetLawsConfirmed { get; set; }
        public string? CustomsNotes { get; set; }
        public RiskLevel RiskLevel { get; set; }
        
        public Shipment Shipment { get; set; } = null!;
    }
} 

File: frontend/src/types/CustomsDeclaration.ts
export interface CustomsDeclaration {
  id?: number;
  shipmentId: string;
  containsLifeforms: boolean;
  lifeformType?: string;
  isPlasmaActive: boolean;
  plasmaStabilityLevel?: number;
  quarantineRequired: boolean;
  originPlanetLawsConfirmed: boolean;
  customsNotes?: string;
  riskLevel: number;
}

export type RiskLevelString = 'Low' | 'Medium' | 'High' | 'Critical';

export const mapRiskLevelToString = (level: number): RiskLevelString => {
  switch(level) {
    case 0: return 'Low';
    case 1: return 'Medium';
    case 2: return 'High';
    case 3: return 'Critical';
    default: return 'Low';
  }
};

export const mapStringToRiskLevel = (level: RiskLevelString): number => {
  switch(level) {
    case 'Low': return 0;
    case 'Medium': return 1;
    case 'High': return 2;
    case 'Critical': return 3;
    default: return 0;
  }
}; 